code/modules/mob/living/carbon/human/life.dm

```diff
@@ -14,9 +14,15 @@
 	//update the current life tick, can be used to e.g. only do something every 4 ticks
 	life_tick++
 
-	if(stat == DEAD)
-		if(!check_tod(src))
-			SSmobs.stop_processing(src)
+	if(stat == DEAD && undefibbable)
+		SSmobs.stop_processing(src) //Last round of processing.
+		if(CHECK_BITFIELD(status_flags, XENO_HOST))
+			var/obj/item/alien_embryo/parasite = locate(/obj/item/alien_embryo) in src
+			if(parasite) //The larva cannot survive without a host.
+				qdel(parasite)
+			DISABLE_BITFIELD(status_flags, XENO_HOST)
+		med_hud_set_status()
+		return TRUE
```

Comment:
Remove the return here

Reply:
```
spookydonut: so remove the return and set medhud there for 100% correct functionality
spookydonut: actually no
spookydonut: make a proc called set_undefibbable
spookydonut: and use that everywhere instead of undefibbale = true
```
No longer needed, it seems.